x-pack/functionbeat/_meta/beat.reference.yml

```diff
@@ -11,9 +11,12 @@
 # Configure functions to run on AWS Lambda, currently we assume that the credentials
 # are present in the environment to correctly create the function when using the CLI.
 #
+# Configure which S3 bucket we should upload the lambda artifact.
+functionbeat.provider.aws.deploy_bucket: "functionbeat-deploy"
+
```

Comment:
I have mixed feeling about how we want to display **provider** options vs **functions**.
How to make it clear in the YAML.

Reply:
I'd say the format chosen very much fits our conventions. If there is a chance we might introduce even more settings you can write:

```
functionbeat.provider.aws:
  deploy_bucket: ...
  <more settings> ?
```

I'm in favor of having `functionbeat.provider.aws.functions:` top-level. This reduces the amount of indentation and copy'n paste errors.

Alternatively:

```
functionbeat.providers.aws:
    deploy_bucket: ...

functionbeat.functions:
- provider: aws
  type: cloudwatch_logs
  ...

```

I wonder if we even need to have an explicit provider on functions. Normally the provider is implicit by function type. By having providers 'implicit', we can't limit the max number of providers in a config file to 1.